src/Orleans/Runtime/GrainReference.cs

```diff
@@ -15,6 +15,12 @@ public class GrainReference : IAddressable, IEquatable<GrainReference>, ISeriali
         private readonly string genericArguments;
         private readonly GuidId observerId;
 
+        /// <summary>
+        /// Invoke method options specific to this grain reference instance
+        /// </summary>
+        [NonSerialized]
+        private readonly InvokeMethodOptions invokeMethodOptions;
```

Comment:
If it woun't be copied by that constryctor - the will be no other way to set it besides making ```invokeMethodOptions``` internal protected property. Also, reference can escape even without that constructor by assingnin it to some outer campured reference. But there's always exists some ways of shooting in the leg, don't think can block them all. Also such escaping will allow reuse one-way enabled grain reference.